app/models/user_subject.rb

```diff
@@ -26,9 +26,59 @@ class UserSubject < ApplicationRecord
               allow_nil: true
 
   # Scopes
-  scope :active, -> {where(finished_at: nil)}
-  scope :completed, -> {where.not(finished_at: nil)}
+  scope :active, -> {where(completed_at: nil)}
+  scope :completed, -> {where.not(completed_at: nil)}
+  scope :for_course, lambda {|course|
+    joins(:course_subject).where(course_subject: {course:})
+  }
```

Comment:
```suggestion
  scope :for_course, (lambda do |course|
    joins(:course_subject).where(course_subject: {course:})
  end)
```
lambda nhiều dòng thì dùng do end e nha